authors/apps/authentication/tests/test_base.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework.test import APIClient, APITestCase
+
+from ..models import User
+from ..serializers import RegistrationSerializer
+
+
+class BaseViewTest(APITestCase):
+    
+    client = APIClient()
+
+    def register_user(self, username="", email="", password=""):
+
+        url = '/api/users/'
+        user_credentials = {
+            "user":{
+                "username": username, 
+                "email": email, 
+                "password": password
+                }
+            }
+        response = self.client.post(url, user_credentials, format="json")
+
+        return response
+
+
+    def log_in_user(self, email="", password=""):
+
+        url = '/api/users/login/'
+        user_credentials = {
+            "user":{ 
+                "email": email, 
+                "password": password
+                }
+            }
+        response = self.client.post(url, user_credentials, format="json")
+
+        return response
+
+
+    def setUp(self):
+        # register user
+        self.register_user("simuser", "simulations.sims@andela.com", "simulman101")
+
+        # create user instance
+        self.user = User.objects.create(
+            username="jealous", 
+            email="simulationtys.sims@andela.com", 
+            password="simulman2101"
+            )
+        
+    
+    
```

Comment:
Always had a new line at the end of the file. You can configure your editor to do that for you.